reportsExport.php

```diff
@@ -295,6 +307,86 @@ function exportDonorsLessThanTwoYears() {
     //exit();
 }
 
+// Export Function for the Report on Donor's who's Frequncy of Giving is Greater than Yearly
+function exportDonorsFOGGTY() {
+    include_once('database/dbinfo.php'); // Make sure you have your database connection setup here
+    $connection = connect();  // This should be your function to establish a database connection
+    
+    // Your SQL query to fetch the required data
+    $query = "SELECT d.Email, p.FirstName, p.LastName, p.PhoneNumber, COUNT(d.email) AS Number_Of_Donations, 
+                    DATEDIFF( CURRENT_DATE(), MIN(DateOfContribution)) AS DateDiff  
+                    FROM dbdonations AS d
+                    JOIN dbdonors AS p ON d.Email = p.Email
+                    GROUP BY d.Email";
+    $result = mysqli_query($connection, $query);
+	
+    header('Content-Type: text/csv');
+    header('Content-Disposition: attachment; filename="donors_Frequncy_Of_Giving.csv"');
+    
+    $output = fopen("php://output", "w");
+    
+    // Write the CSV header
+    fputcsv($output, array('Email', 'First Name', 'Last Name', 'Phone Number', 'Frequency of Giving', 'Days From Earliest Donation'));
+    
+    // Write rows
+    while ($row = mysqli_fetch_assoc($result)) {
+		$formattedPhone = '(' . substr($row['PhoneNumber'], 0, 3) . ') ' . substr($row['PhoneNumber'], 3, 3) . '-' . substr($row['PhoneNumber'], 6);
+		
+		// Frequency of Giving
+		$FOG = "";
+		$Rate = 0;
+		$ratio = $row['Number_Of_Donations'] / ($row['DateDiff'] / 365);
+		if ($ratio < 1){
+			$FOG="Less Than Yearly";
+		} elseif ($ratio <6 && $ratio >= 1){
+			$FOG = "Yearly";
+		} elseif ($ratio >= 6 && $ratio < 12){ // Either comment this out for now or remove it since bimonthly isn't needed
+			$FOG = "Bi-Monthly";
+			$Rate = 1;
+		} elseif ($ratio >= 12){
+			$FOG = "Monthly";
+			$Rate = 1;
+		}
+		if ($Rate == 1){
+		fputcsv($output, array($row['Email'], $row['FirstName'], $row['LastName'], $formattedPhone, $FOG, $row['DateDiff']));
+		}
+	}
+    fclose($output);
+}
+// Export Function for the Report on Donor's in the Past Three Years
```

Comment:
well documented code for function for donors who have donated in the past 3 years.